### ISSUE TITLE
RandomForestLearning : Examples can't run with SampleRatio not equal 1.0 

### ISSUE THREAD

**Issue by khanhjp** [accord-net/framework#576]
http://accord-framework.net/docs/html/T_Accord_MachineLearning_DecisionTrees_RandomForestLearning.htm

ERROR :
```
- forest.Decide (inputsPattern); 'forest.Decide (inputsPattern)' threw an exception of type 'System.AggregateException'. Int [] {System.AggregateException}
- base Count = 1 System.Exception {System.AggregateException}
+ Data {System.Collections.ListDictionaryInternal} System.Collections.IDictionary {System.Collections.ListDictionaryInternal}
HelpLink null string
 + InnerException {"Index is out of bounds of array"} System.Exception** {System.IndexOutOfRangeException}
Message "One or more errors occurred" string**
Source "mscorlib" string
StackTrace "location System.Threading.Tasks.Task.ThrowIfExceptional (Boolean includeTaskCanceledExceptions) \ r \ n Location System.Threading.Tasks.Task.Wait (Int32 millisecondsTimeout, CancellationToken cancellationToken) \ r \ n Location System.Threading.Tasks.Parallel. ForWorker [TLocal] (Int32 fromInclusive, Int32 toExclusive, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Func`4 bodyWithLocal, Func`1 localInit, Action`1 localFinally) \ r \ n Location System.Threading.Tasks. Place Accord.MachineLearning.DecisionTrees.RandomForest.Decide (Double [] input) \ r \ n Location Accord.MachineLearning.ClassifierBase` 2 (Int32 to Inclusive, Int32 to Exclusive, ParallelOptions parallelOptions, Action`1 body) .Decide (TInput [] input, TClasses [] result) \ r \ n Location Accord.MachineLearning.ClassifierBase`2.Decide (TInput [] input) "string
+ TargetSite {Void ThrowIfExceptional (Boolean)} System.Reflection.MethodBase {System.Reflection.RuntimeMethodInfo}
+ Static members
+ Non-public members
+ InnerExceptions Count = 1 System.Collections.ObjectModel.ReadOnlyCollection <System.Exception>
+ Non-public members
```
Thanks for any and all help!

**Comment by khanhjp** [comment]
Can no one help me?

**Comment by cesarsouza** [comment]
Apologies for the delay. At the time the issue was first opened I couldn't investigate it right away and the issue ended up being moved to the backlog for a long time. It should have been fixed in the latest release (3.8.0).
